cmd/bclient/bserver/bserver.go

```diff
@@ -0,0 +1,118 @@
+package bserver
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/antonholmquist/jason"
+	"github.com/ndlib/bendo/cmd/bclient/fileutil"
+	"sync"
+)
+
+var (
+	ItemFetchStatus error
+
+	fileIDMutex sync.Mutex
+
+	fileIDList []fileIDStruct
+
+	bendoServer *string
+	RemoteJason *jason.Object
+)
+
+type fileIDStruct struct {
+	fileid string
+	slot   string
+	item   string
+}
+
+func addFileToTransactionList(filename string, fileID string, item string) {
+
+	fileIDMutex.Lock()
+
+	thisFileID := new(fileIDStruct)
+	thisFileID.fileid = fileID
+	thisFileID.slot = filename
+	thisFileID.item = item
+
+	fileIDList = append(fileIDList, *thisFileID)
+
+	fileIDMutex.Unlock()
+}
+
+func Init(server *string) {
+	fileutil.IfVerbose("github.com/ndlib/bendo/bclient/bserver.Init() called")
+	bendoServer = server
+}
+
+func FetchItemInfo(item string) {
+	defer fileutil.UpLoadDone.Done()
+	fileutil.IfVerbose("github.com/ndlib/bendo/bclient/bserver.FetchItemInfo() called")
+
+	json, err := GetItemInfo(item)
+
+	// Some error occurred retrieving from server, or item not found.
+	if err != nil {
+		ItemFetchStatus = err
+		fmt.Println(err.Error())
+		return
+	}
+
+	RemoteJason = json
+}
+
+// serve the file queue. This is called from main as 1 or more goroutines
+// If the file Upload fails, close the channel and exit
+
+func SendFiles(fileQueue chan string, item string, fileroot string, mut *sync.WaitGroup) {
+
+	for filename := range fileQueue {
+		err := uploadFile(filename, fileutil.ShowUploadFileMd5(filename), item, fileroot)
+
+		if err != nil {
+			close(fileQueue)
+		}
+	}
+
+	mut.Done()
+}
+
+func uploadFile(filename string, uploadMd5 []byte, item string, fileroot string) error {
+
+	// upload chunks initial buffer size is 1MB
+
+	fileID, uploadErr := chunkAndUpload(fileroot, filename, uploadMd5, item, 1048576)
```

Comment:
It must be my message passing history :)


On Fri, Jan 22, 2016 at 1:57 PM, Don Brower <notifications@github.com>
wrote:

> In cmd/bclient/bserver/bserver.go
> <https://github.com/ndlib/bendo/pull/56#discussion_r50575544>:
>
> > +	for filename := range fileQueue {
> > +		err := uploadFile(filename, fileutil.ShowUploadFileMd5(filename), item, fileroot)
> > +
> > +		if err != nil {
> > +			close(fileQueue)
> > +		}
> > +	}
> > +
> > +	mut.Done()
> > +}
> > +
> > +func uploadFile(filename string, uploadMd5 []byte, item string, fileroot string) error {
> > +
> > +	// upload chunks initial buffer size is 1MB
> > +
> > +	fileID, uploadErr := chunkAndUpload(fileroot, filename, uploadMd5, item, 1048576)
>
> the same variables seem to be passed from SendFiles to uploadFile and
> then to chunkAndUpload. Maybe make some of those variables structure
> members and turn these three function into methods on that structure?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ndlib/bendo/pull/56/files#r50575544>.
>